analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java

```diff
@@ -644,4 +711,37 @@ private void writeRepositorySettingsFile(Map<ArtifactRepository, Path> repositor
         return allDependencies;
     }
 
+    public void writeReport(File rootDir, String report) throws ManipulationException {
+        final File reportFile = new File(".", "report-pme-"
+                + RandomStringUtils.randomAlphanumeric(10) + ".txt");
```

Comment:
I don't think using `randomAlphanumeric` is correct - we had already agreed a naming format and this makes it unpredictable. We should review where PME puts the files - putting them in a non-build directory may mean repour commits them which might not be desirable.

Reply:
I know it is not correct.

If I use `project.getRootDir()`, I don't find the files at all in the directory tree. I wanted to check what `project.getRootDir()` is being set to, but I cannot see any logging output from the tests.

If I set to `.`, then it adds it in the source tree, which means the files overwrite themselves on the next test and I couldn't see them. This is the reason for adding the random component just for now.

I would have hoped that tests would run outside of the source tree as well. Creating files inside the source tree is not right. For example, maven uses `target/`, then the build files don't pollute the source tree.

The files are always created here, but in PME, the report txt is optional, and doesn't appear to have an official name. It looks like the default is just empty string `""`. I don't really see equivalent options in GME. But, does this file have a standard name?

Reply:
There are a lot of separate comments here which probably should be discussed.

I think the output should go in the build directory. But even then, repeated runs, would overwrite (which I would expect).